packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -519,19 +535,52 @@ export const DataProviderMixin = (superClass) =>
       return result;
     }
 
-    scrollToIndex(index) {
-      super.scrollToIndex(index);
-      if (!isNaN(index) && (this._cache.isLoading() || !this.clientHeight)) {
-        this.__pendingScrollToIndex = index;
+    /**
+     * Scroll to a specific row index in the virtual list. Note that the row index is
+     * not always the same for any particular item. For example, sorting/filtering/expanding
+     * or collapsing hierarchical items can affect the row index related to an item.
+     *
+     * The `indexes` parameter can be either a single number or multiple numbers.
+     * The grid will first try to scroll to the item at the first index on the top level.
+     * In case the item at the first index is expanded, the grid will then try scroll to the
+     * item at the second index within the scope of the expanded first item, and so on.
+     * Each given index points to a sub-item of the previous index.
+     *
+     * @param indexes {number|number[]} the index of the item
+     */
+    scrollToIndex(...indexes) {
+      this._scrollToIndex(this.__getGlobalFlatIndex(indexes));
```

Comment:
This doesn't seem to work if a sub-cache hasn't been initialized yet.

Let's say there are 200 items on each level. Then run:
```
grid.scrollToIndex(75, 100);
```

Currently that only scrolls to `75`.

Reply:
This was intentional. I wanted to avoid magic as much as possible (auto-expanding items). That's another thing to discuss perhaps.

Reply:
Ah sorry, I have expanded the item before scrolling:
```
grid.expandedItems = [{ name: '75' }];
```

Also noting that scrolling a second time after the sub-items have been loaded works.

Reply:
Another interesting scenario: Exand one of the items just before the one that was scrolled to:
```
grid.expandedItems = [{ name: '74' }];
grid.scrollToIndex(75, 100);
```
You'll see the child items for `74` instead.

Reply:
Tested the first case. It seems to work as expected.

I'm guessing you're using the grid in the new test block? The test setup stores references to the data provider callbacks and they need to be flushed after the grid asks for data. You can alternatively change it to use timeouts instead: `setTimeout(() => cb(pageItems, levelSize), 100);`;

Reply:
Pushed a test case in https://github.com/vaadin/web-components/pull/5428/commits/7df270b786b098e9e908c3c68689dd89b7416aa5

Reply:
I'm using a modified dev page for testing:
```html
<script type="module">
  import "@vaadin/grid/all-imports";

  const grid = document.querySelector("vaadin-grid");
  grid.dataProvider = ({ parentItem, page, pageSize }, cb) => {
    // Let's have 100 root-level items and 5 items on every child level
    const levelSize = parentItem ? 200 : 200;

    const pageItems = [...Array(Math.min(levelSize, pageSize))].map((_, i) => {
      const indexInLevel = page * pageSize + i;

      return {
        name: `${parentItem ? parentItem.name + "-" : ""}${indexInLevel}`,
        children: true
      };
    });

    cb(pageItems, levelSize);
  };

  grid.expandedItems = [
    { name: "75" }
  ];

  document.getElementById("scroll").addEventListener("click", e => {
    grid.scrollToIndex(75, 100);
  });
</script>
<vaadin-grid item-id-path="name">
  <vaadin-grid-tree-column path="name" width="200px" flex-shrink="0"></vaadin-grid-tree-column>
</vaadin-grid>
<button id="scroll">Scroll</button>
```

Reply:
> Another interesting scenario:

Works for me as expected. Pushed a test case for this also https://github.com/vaadin/web-components/pull/5428/commits/312bf2be7020186cc2f5302219d683dbc5f09d27

Reply:
It works when adding a timeout to the callback, so it looks like the cache loading check is missed when the callback is synchronous.